projects/ui/option/option.ts

```diff
@@ -99,9 +103,13 @@ export class CkOption<T = unknown> implements Highlightable, FocusableOption {
     public readonly host: ElementRef<HTMLElement>,
   ) {}
 
+  public setRole(role: string): void {
+    if (!this.role()) this._role.set(role)
+  }
+
   /** Selects the option. */
   public select(emitEvent = true): void {
-    if (this.disabled) return
+    if (this.isSelected() || this.disabled) return
```

Comment:
Was there any particular reason for adding this condition?

Reply:
Just tried to keep consistently. We have the similar approach in `deselect`, so I added the same. If we select the option we don't need to set the same value again and emit an event. It's a common practice.

Reply:
Yea, that's clear, but I recall there was specific reason I didn't do that initially, because I sill wanted to be notified about selection in Autocomplete. Just make sure it didn't break anything there

Reply:
You will be notified if you select a new option and you won't if you select already selected. All good.